processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -453,7 +453,10 @@ TimeAndDims toTimeAndDims(InputRow row) throws IndexSizeExceededException
       dims = newDims;
     }
 
-    long truncated = gran.truncate(row.getTimestampFromEpoch());
+    long truncated = 0;
+    if (row.getTimestamp() != null) {
```

Comment:
[NoneGranularity.truncate()](https://github.com/druid-io/druid/pull/3850/files/27679a25f580b914e180314b6d60aac2ce7c2f43#diff-358e35786ef6a6927a1723828d267ce1R50) returns a null when its input is null. I believe I ran into such scenarios and that's why this was added.